### PR TITLE
HHH-12349 - User Guide documentation for @Filter is too verbose

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-12349

### CODE REVIEW

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc [hunk 1]
```diff
@@ -1829,19 +1829,40 @@ include::{sourcedir}/basic/WhereJoinTableTest.java[tags=mapping-where-join-table
...
+include::{sourcedir}/basic/FilterTest.java[tags=mapping-filter-Account-example]
 ----
 ====
 
-If the database contains the following entities:
+[NOTE]
+====
+Notice that the `active` property is mapped to the `active_status` column.
+
+This mapping was map to show you that the `@Filter` condition uses a SQL condition, and not a JPQL filtering criteria.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: `was map`?

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc [hunk 2]
```diff
@@ -1829,19 +1829,40 @@ include::{sourcedir}/basic/WhereJoinTableTest.java[tags=mapping-where-join-table
...
-.`@Filter` mapping usage
+Now, considering we have the following `Account` entity:
+
+[[mapping-filter-account-example]]
+.`@Filter` mapping entity-level usage
 ====
 [source, JAVA, indent=0]
 ----
-include::{sourcedir}/basic/FilterTest.java[tags=mapping-filter-example]
+include::{sourcedir}/basic/FilterTest.java[tags=mapping-filter-Account-example]
```
Comment (gsmet) [inline documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: I don't know in ORM but in other projects, we usually keep the tags all lowercase. Please ignore if it's not the case in ORM.

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc [hunk 3]
```diff
@@ -1829,19 +1829,40 @@ include::{sourcedir}/basic/WhereJoinTableTest.java[tags=mapping-where-join-table
...
+====
+
+As already explained, we can also apply the `@Filter` annotation for collections as illustrated by the `Client` entity:
+
+[[mapping-filter-client-example]]
+.`@Filter` mapping collection-level usage
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/FilterTest.java[tags=mapping-filter-Client-example]
```
Comment (gsmet) [inline documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: Same comment about the tag here.

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc [hunk 4]
```diff
@@ -1945,7 +1997,11 @@ include::{sourcedir}/basic/FilterJoinTableTest.java[tags=mapping-filter-join-tab
 ----
 ====
 
-If the database contains the following entities:
+The `firstAccounts` filter will allow us to get only the `Account` entities that have the `order_id`
+(which tels the position of every entry inside the `accounts` collection)
```
Comment (gsmet) [inline documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: s/tels/tells/

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc [hunk 5]
```diff
@@ -1945,7 +1997,11 @@ include::{sourcedir}/basic/FilterJoinTableTest.java[tags=mapping-filter-join-tab
...
+less than a given number (e.g. `maxOrderId`)
```
Comment (gsmet) [inline documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: Missing dot at the end of the sentence.

**Comment by gsmet** [comment]
And +1 to push it to 5.3 too.

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc (hunk 1 above)
Comment (vladmihalcea) [reply documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: "was done" makes more sense.

File: documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc (hunk 2 above)
Comment (vladmihalcea) [reply documentation/src/main/asciidoc/userguide/chapters/domain/basic_types.adoc]: I think we already used this pattern. We could address it using a separate issue.

**Pull request merged**